### PR TITLE
Adding Notification to TG-bot

### LINKED ISSUE #694: Add Notification to Telegram bot

### PR DESCRIPTION
**Pull request by Tkachelnik**

### CODE REVIEW

File: CharlieBackend.Business/Services/Notification/HangfireJobService.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            var hangfireJobIds = await _unitOfWork.JobMappingRepository.GetHangfireIdsByCustomJobId(customJobId);
+
+            foreach (var Id in hangfireJobIds)
+            {
+                BackgroundJob.Delete(Id);
+            }
+
+            var jobId = BackgroundJob.Enqueue<IHangfireNotificationService>(x => x.CreateAddHomeworkStudentNotification(homeworkStudent.Id));
+
+            return jobId != null ? true : false;
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/Notification/HangfireJobService.cs]: `return jobId != null;` - that is enough.

File: CharlieBackend.Business/Services/Notification/HangfireJobService.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        public async Task<bool> CreateUpdateHomeworkStudentJob(HomeworkStudent homeworkStudent)
+        {
+            var jobId = BackgroundJob.Enqueue<IHangfireNotificationService>(x => x.CreateUpdateHomeworkStudentNotification(homeworkStudent.Id));
+
+            return jobId != null ? true : false;
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/Notification/HangfireJobService.cs]: `return jobId != null;` - that is enough.

File: CharlieBackend.Business/Services/Notification/HangfireJobService.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            var hangfireJobIds = await _unitOfWork.JobMappingRepository.GetHangfireIdsByCustomJobId(customJobId);
+
+            foreach (var Id in hangfireJobIds)
+            {
+                BackgroundJob.Delete(Id);
+            }
+
+            var jobId = BackgroundJob.Enqueue<IHangfireNotificationService>(x => x.CreateUpdateMarkNotification(homeworkStudent.Id));
+
+            return jobId != null ? true : false;
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/Notification/HangfireJobService.cs]: `return jobId != null;` - that is enough.

File: CharlieBackend.Data/ApplicationContext.cs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿using CharlieBackend.Core.Entities;
+﻿ using CharlieBackend.Core.Entities;
```
Comment (andrii-developer) [inline CharlieBackend.Data/ApplicationContext.cs]: Please remove the space.

File: CharlieBackend.Root/CompositionRoot.cs [hunk 5]
```diff
@@ -1,30 +1,51 @@
...
         public static void InjectDependencies(IServiceCollection services, IConfiguration configuration)
         {
             services.AddDbContext<ApplicationContext>(options =>
             {
                 options.UseMySql(configuration.GetConnectionString("DefaultConnection"),
-                  b => b.MigrationsAssembly("CharlieBackend.Api"));
+                  b =>
+                  {
+                      b.MigrationsAssembly("CharlieBackend.Api");
+                      //b.EnableRetryOnFailure(5, TimeSpan.FromSeconds(10), null);
```
Comment (andrii-developer) [inline CharlieBackend.Root/CompositionRoot.cs]: Please remove the commented code.

File: WhatBackend.TelegramBotApi/Controllers/MessageController.cs [hunk 6]
```diff
@@ -29,7 +29,11 @@ public string Get()
         [HttpPost]
         public async Task<IActionResult> Update([FromBody] Update update)
         {
-           var client = await Bot.Get(_serviceProvider);
+            if(DateTime.Now >= new DateTime(2021,10,13,17,30, 1))
```
Comment (andrii-developer) [inline WhatBackend.TelegramBotApi/Controllers/MessageController.cs]: What is the purpose of that code?

File: WhatBackend.TelegramBotApi/Startup.cs [hunk 7]
```diff
@@ -51,24 +52,22 @@ public void ConfigureServices(IServiceCollection services)
             services.AddControllers().AddNewtonsoftJson();
         }
 
+        private DbConnection ApiSettings()
```
Comment (andrii-developer) [inline WhatBackend.TelegramBotApi/Startup.cs]: What is the purpose of that method?

**Review by andrii-developer — CHANGES_REQUESTED** [review]

**Review by andrii-developer — APPROVED** [review]

**Review by Ilhamich — APPROVED** [review]

**Review by DmitriyFesak — APPROVED** [review]

**Review by Q-Alex-Q — APPROVED** [review]

**Pull request merged**